agent/engine/docker_task_engine_test.go

```diff
@@ -1146,3 +1146,27 @@ func TestEngineDisableConcurrentPull(t *testing.T) {
 	assert.False(t, dockerTaskEngine.enableConcurrentPull,
 		"Task engine should not be able to perform concurrent pulling for version < 1.11.1")
 }
+
+// TestTaskWithCircularDependency tests the task with containers of which the
```

Comment:
Can you also add a unit test in `dependencygraph` for `ValidDependencies()` to test a task with circular dependencies?